### PR TITLE
Suggestion - Auto menu placement

### PR DESCRIPTION
**Pull request by aspirisen**
Suggestion to make menu adaptive to the screen size

### CODE REVIEW

File: packages/slate-editor/src/components/EditorBlock/Menu.tsx [hunk 1]
```diff
@@ -108,37 +112,39 @@ export class Menu extends Component<Props> {
...
+                                    [styles.top]: placement.indexOf('top') >= 0,
+                                    [styles.bottom]: placement.indexOf('bottom') >= 0,
+                                    [styles.left]: placement.indexOf('left') >= 0,
+                                    [styles.right]: placement.indexOf('right') >= 0,
+                                })}
+                                {...arrowProps}
+                            />
+                            {children}
+                        </Toolbox.Panel>,
+                        document.getElementById('full-page-react-application') as HTMLElement,
```
Comment (aspirisen) [inline packages/slate-editor/src/components/EditorBlock/Menu.tsx:145]: There should be some container from props

Comment (kudlajz) [reply packages/slate-editor/src/components/EditorBlock/Menu.tsx:145]: I think we could add the prop into the context I've introduced.

File: packages/slate-editor/src/components/EditorBlock/Menu.module.scss [hunk 2]
```diff
@@ -1,7 +1,10 @@
+@import "styles/variables";
+
 .menu {
     width: 280px;
     position: absolute;
     backdrop-filter: blur(6px);
+    z-index: $toolbar-z-index;
```
[the remark this answers is not in the document]
Comment (aspirisen) [reply packages/slate-editor/src/components/EditorBlock/Menu.module.scss:7]: yeah, it looks like the best place, or just explicit prop on the editor to control editor styles (maybe we will have some cases where we need to control styles as well). 
It is up to you, i have no strong opinion on how to provide this z-index

File: packages/slate-editor/src/components/EditorBlock/Menu.tsx (hunk 1 above)
Comment (kudlajz) [reply packages/slate-editor/src/components/EditorBlock/Menu.tsx:145]: I was able to make it work just by providing `document.body` here. Maybe we can use that for now and add the prop in the future when/if needed?

**Review by kudlajz — APPROVED** [review]
I've used `document.body` for the portal and added support for settings the z-index via the popper options.

File: packages/slate-editor/src/components/EditorBlock/Menu.tsx (hunk 1 above)
Comment (aspirisen) [reply packages/slate-editor/src/components/EditorBlock/Menu.tsx]: Yeah, let's leave for now. The one problem could be if user's browser extensions will mutate body's elements, but it is an edge case for now

**Pull request merged**